### DISCUSSION TITLE
Use of hipMalloc in the "Finite difference method" labs

### DISCUSSION THREAD

**Discussion by AlexisEspinosaGayosso** [amd/amd-lab-notes#11] [Q&A] [answered]
Why are you using:
`
hipMalloc((void**)&d_u, numbytes);
hipMalloc((void**)&d_f, numbytes);
`

instead of the "typical":
`
hipMalloc(&d_u, numbytes);
hipMalloc(&d_f, numbytes);
`

?

OK. Got it. `hipMalloc` receives, as first argument a `void **ptr`. Then, the `(void **)&d_u` is performing an explicit cast to the type definition of that first argument, rather than letting the compiler perform the casting. I guess the decision of using an explict cast is more a matter of style. Or is there any "AMD GPU guru" hint that make you decide to use the explicit cast?

**Comment by jychang48** [comment, marked as answer]
@AlexisEspinosaGayosso yes, this was simply a matter of style, the former is compatible with both C and C++ whereas the latter is only compatible with C++, but since the provided example is written in C++, the latter should be sufficient.
